pkg/controller/manager/node/controller.go

```diff
@@ -117,3 +134,21 @@ func (h Handler) ensureMgmtLabels(node *corev1.Node) error {
 
 	return nil
 }
+
+// Clear link statuses related to the removed node
+func (h Handler) clearLinkStatus(nodeName string) error {
+	lms, err := h.lmCache.List(labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	for _, lm := range lms {
+		lmCopy := lm.DeepCopy()
+		delete(lmCopy.Status.LinkStatus, nodeName)
```

Comment:
Check if the link status exists before deleting it to avoid trying to delete a non-existing link status. e.g.,

```golang
if _, ok := lmCopy.Status.LinkStatus[nodeName]; ok {
    delete(lmCopy.Status.LinkStatus, nodeName)
    // update the linkstatus
}